app/src/androidTest/java/at/tugraz/ist/guessingwords/data/service/WordServiceTest.kt

```diff
@@ -128,4 +130,109 @@ class WordServiceTest {
+                finished.open()
+            }
+        }
+        service.createNewMultiplayerWordPool(cbInMemory)
+        cbInMemory.finished.block()
+        //Merge Some words
+
+        val cbMerge = object:Callback<List<Long>>{
+            val finished = ConditionVariable()
+            var resWordIds = listOf<Long>()
+            override fun whenReady(data: List<Long>?) {
+                resWordIds = data!!
+                finished.open()
+            }
+        }
+        val wordList = listOf(Word("test1"), Word("test2"), Word("test3"))
+        service.mergeIntoDatabase(wordList, cbMerge)
+        cbMerge.finished.block()
+
+        // get all Words test
+        val cbGetAll = object:Callback<List<Word>>{
+            val finished = ConditionVariable()
+            var resWords = listOf<Word>()
+            override fun whenReady(data: List<Word>?) {
+                resWords = data!!
+                finished.open()
+            }
+        }
+        service.getAllWords(cbGetAll)
+        cbGetAll.finished.block()
+        assert(cbGetAll.resWords.count() >= 3)
+        service.removeMultiplayerWordPool()
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    @Test(timeout = 3000)
+    fun checkForDuplicatesAfterMerge(){
+        val word = Word("test1")
+        val cb = object:Callback<Long>{
+            val finished = ConditionVariable()
+            override fun whenReady(data: Long?) {
+                finished.open()
+            }
+        }
+        service.insertOrUpdateExistingWord(word, cb)
+        cb.finished.block()
+
+        // In memory Instance create
+        val cbInMemory = object:Callback<List<Long>>{
+            val finished = ConditionVariable()
+            override fun whenReady(data: List<Long>?) {
+                finished.open()
+            }
+        }
+        service.createNewMultiplayerWordPool(cbInMemory)
+        cbInMemory.finished.block()
+        //Merge Some words
+
+        val cbMerge = object:Callback<List<Long>>{
+            val finished = ConditionVariable()
+            override fun whenReady(data: List<Long>?) {
+                finished.open()
+            }
+        }
+        val wordList = listOf(Word("test2"), Word("test1"), Word("test3"))
+        service.mergeIntoDatabase(wordList, cbMerge)
+        cbMerge.finished.block()
+
+        // get all Words test
+        val cbGetAll = object:Callback<List<Word>>{
+            val finished = ConditionVariable()
+            var resWords = listOf<Word>()
+            override fun whenReady(data: List<Word>?) {
+                resWords = data!!
+                finished.open()
+            }
+        }
+        service.getAllWords(cbGetAll)
+        cbGetAll.finished.block()
+        cbGetAll.resWords.forEach { word ->
+            if (cbGetAll.resWords.count{ x -> x.text.lowercase() == word.text.lowercase() } > 1){
+                assert(false)
+            }
+        }
+        assert(true)
```

Comment:
Firstly, asserting true does nothing.
Secondly, because you can control exactly which input to expect here, I would instead recommend an assert that is easier to read and constructed exactly for this test.
Eg. `assert(cbGetAll.resWords.size == 3)` or `assert(cbGetAll.resWords.filter { word -> word.text.toLowerCase() == "test1" }.size == 1)` or both.